File: LeetCode/Top Interview Questions/Array/Easy/intersection_of_2_arrays_2_hari.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map <int ,int> n1,n2;
        int i;
        for(i=0;i<nums1.size();i++)
        {
            if(n1.find(nums1[i]) == n1.end())
            {
                n1[nums1[i]]=1;
            }
            else
            {
                n1[nums1[i]]++;
            }
        }
        for(i=0;i<nums2.size();i++)
        {
            if(n2.find(nums2[i]) == n2.end())
            {
                n2[nums2[i]]=1;
            }
            else
            {
                n2[nums2[i]]++;
            }
        }
        vector<int> result;
        int n,j;
        for (auto i : n1)
        {
            if(n2.find(i.first) != n2.end())
            {
                //cout<<i.first<<' '<<i.second<<' '<<n2[i.first]<<endl;
                n=min(i.second , n2[i.first]);
                for(j=0;j<n;j++)
                {
                    result.push_back(i.first);
                }
            }
            // for(j=0;j<n;j++)
            // {
            //     result.push_back(i.first);
            // }
        }
        return result;
    }
};

File: LeetCode/Top Interview Questions/Array/Easy/intersection_of_2arrays2_hari.py
 class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        count = {}
        d = {}

        for i in nums1:
            if i in count:
                count[i] += 1
            else:
                count[i] = 1
            d[(i, count[i])] = 1
        #print(d)
        #print(count)
        count = {}
        for i in nums2:
            if i in count:
                count[i] += 1
            else:
                count[i] = 1
            if (i, count[i]) in d:
                d[(i, count[i])] += 1
        #print(d)
        vals = []
        for key in d:
            #print(vals)
            if d[key] > 1:
                vals.append(key[0])

        return vals



class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        hash1={}
        hash2={}
        for i in nums1:
            if i in hash1:
                hash1[i]+=1
            else:
                hash1[i]=1
        for i in nums2:
            if i in hash2:
                hash2[i]+=1
            else:
                hash2[i]=1
        res = []
        for i in hash1:
            if i in hash2:
                v=min(hash1[i],hash2[i])
                while v:
                    res.append(i)
                    v-=1
        return res
